File: apps/dsa-web/src/types/notifications.ts
/** 单条通知投递回执（camelCase，由 api/notifications 的 snake->camel mapper 构建）。 */
export interface NotificationDeliveryItem {
  id: number;
  routeType: string;
  channel: string;
  attempt: number;
  success: boolean;
  errorCode: string | null;
  retryable: boolean;
  latencyMs: number | null;
  diagnostics: string | null;
  createdAt: string;
}

export interface NotificationDeliveryListResponse {
  items: NotificationDeliveryItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface NotificationDeliveryListQuery {
  routeType?: string;
  channel?: string;
  success?: boolean;
  page?: number;
  pageSize?: number;
}

File: api/v1/schemas/notifications.py
# -*- coding: utf-8 -*-
"""Schemas for generic (non-alert) notification delivery receipts."""

from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class NotificationDeliveryItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    route_type: str
    channel: str
    attempt: int
    success: bool
    error_code: Optional[str] = None
    retryable: bool
    latency_ms: Optional[int] = None
    diagnostics: Optional[str] = None
    created_at: datetime


class NotificationDeliveryListResponse(BaseModel):
    items: List[NotificationDeliveryItem]
    total: int
    page: int
    page_size: int
